framework/auth/views.py

```diff
@@ -185,6 +193,34 @@ def auth_logout(redirect_url=None):
     return resp
 
 
+def auth_email_logout(token, user):
+    """When a user is adding an email or merging an account, add the email to the user and log them out.
+    """
+    redirect_url = web_url_for('auth_login') + '?existing_user={}'.format(urllib.quote_plus(user.email))
+    try:
+        unconfirmed_email = user.get_unconfirmed_email_for_token(token)
+    except InvalidTokenError:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': "Bad token",
+            'message_long': "The provided token is invalid."
+        })
+    except ExpiredTokenError:
+        status.push_status_message('The private link you used is expired.')
+        return
```

Comment:
This implicitly returns `None`, which seems incorrect, because this means the view will return `None` here: https://github.com/CenterForOpenScience/osf.io/pull/5528/files#diff-9a64b4982d0bd80f7a498c65ddce2023R241 . Probably better to raise an `HTTPError` so that the user sees an error page.